code/game/objects/items/props/helmetgarb.dm

```diff
@@ -94,110 +116,349 @@
+		return
+	if(shape == NVG_SHAPE_COSMETIC)
+		to_chat(user, SPAN_WARNING("There is no connector for the power cell inside \the [src]."))
+		return
+	if(shape == NVG_SHAPE_BROKEN)
+		to_chat(user, SPAN_WARNING("You need to repair \the [src] first."))
+		return
+	if(nvg_charge == nvg_maxcharge)
+		to_chat(user, SPAN_WARNING("\The [src] are already fully charged."))
+		return
 
-/obj/item/prop/helmetgarb/helmet_nvg/proc/toggle_nods(var/mob/user)
-	if(user.is_mob_incapacitated())
+	while(nvg_charge < nvg_maxcharge)
+		if(C.charge <= 0)
+			to_chat(user, SPAN_WARNING("\The [C] is completely dry."))
+			break
+		var/to_transfer
+		if(do_after(user, 1 SECONDS, (INTERRUPT_ALL & (~INTERRUPT_MOVED)), BUSY_ICON_BUILD, C, INTERRUPT_DIFF_LOC))
+			to_transfer = min(400, C.charge, (nvg_maxcharge - nvg_charge))
+			C.charge -= to_transfer
+			nvg_charge += to_transfer
+			to_chat(user, "You transfer some power between \the [C] and \the [src]. The gauge now reads: [round(100.0*nvg_charge/nvg_maxcharge) ]%.")
+		else
+			to_chat(user, SPAN_WARNING("You were interrupted."))
+			break
+
+/obj/item/prop/helmetgarb/helmet_nvg/proc/repair(mob/user as mob)
+	if(user.action_busy)
+
+		return
+	if(src != user.get_inactive_hand())
+		to_chat(user, SPAN_WARNING("You need to hold \the [src] in hand in order to repair them."))
+		return
+	if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_TRAINED)) // level 2 is enough to repair damaged NVG
+		to_chat(user, SPAN_WARNING("You are not trained to repair electronics..."))
 		return
 
+	if(shape == NVG_SHAPE_BROKEN)
+		if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_ENGI)) // level 3 is needed to repair broken NVG
+			to_chat(user, SPAN_WARNING("Repair of this complexity is too difficult for you, find someone more trained."))
+			return
+
+		to_chat(user, "You begin to repair \the [src].")
+		if(do_after(user, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_BUILD, src))
+			playsound(src.loc, 'sound/items/Screwdriver.ogg', 25, 1)
+			src.color = "#bebebe"
+			shape = NVG_SHAPE_PATCHED
+			to_chat(user, "You successfully patch \the [src].")
+			nvg_maxhealth = 65
+			nvg_health = 65
+			nvg_drain = initial(nvg_drain) * 2
+		else
+			to_chat(user, SPAN_WARNING("You were interrupted."))
+
+	else if(nvg_health == nvg_maxhealth)
+		if(shape == NVG_SHAPE_PATCHED)
+			to_chat(user, SPAN_WARNING("Already repaired, nothing more you can do."))
+		else if(shape == NVG_SHAPE_FINE)
+			to_chat(user, SPAN_WARNING("Nothing to fix."))
+		else if(shape == NVG_SHAPE_COSMETIC)
+
+			to_chat(user, SPAN_WARNING("it's nothing but a husk of what it used to be."))
+
+		else
+			to_chat(user, "You begin to repair \the [src].")
+			if(do_after(user, 2 SECONDS, INTERRUPT_ALL, BUSY_ICON_BUILD, src))
+				to_chat(user, "You successfully repair \the [src].")
+				playsound(src.loc, 'sound/items/Screwdriver.ogg', 25, 1)
+				nvg_health = nvg_maxhealth
+			else
+				to_chat(user, SPAN_WARNING("You were interrupted."))
```

Comment:
What is this block supposed to do? If `health == maxhealth`, you can spend time to... make it so that `health = maxhealth`?